### PR TITLE
[bugfix](jsonb memory leak) there are memory leak in jsonb field

### PR DESCRIPTION
**Pull request by yiguolei**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yiguolei** [comment]
run buildall

File: be/src/vec/core/field.h [hunk 1]
```diff
@@ -147,7 +147,12 @@ class JsonbField {
         memcpy(data, x.data, size);
     }
 
-    JsonbField(JsonbField&& x) : data(x.data), size(x.size) {
+    JsonbField(JsonbField&& x) {
+        if (data) {
```
Comment (xiaokang) [inline be/src/vec/core/field.h:151]: It's not necessary for constructor function since the object is not existed yet.

**Comment by yiguolei** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.03 seconds
 stream load tsv:          449 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         26 seconds loaded 2358488459 Bytes, about 86 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230319024413_clickbench_pr_116527.html

**Review by wangbo — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**